mijia/src/bluetooth.rs

```diff
@@ -229,7 +312,153 @@ impl BluetoothSession {
         Ok(sensors)
     }
 
-    fn device(&self, id: &DeviceId) -> impl OrgBluezDevice1 {
+    /// Get a list of all GATT services which the given Bluetooth device offers.
+    ///
+    /// Note that this won't be filled in until the device is connected.
+    pub async fn get_services(
+        &self,
+        device: &DeviceId,
+    ) -> Result<Vec<ServiceInfo>, BluetoothError> {
+        let introspection_xml = self.device(device).introspect().compat().await?;
+        let device_node: Node = serde_xml_rs::from_str(&introspection_xml)?;
```

Comment:
(optional) DBusNode or something?

Reply:
They are in the `introspect` module, which I think is enough of a namespace.